portfolio/src/main/java/com/google/sps/servlets/LogoutServlet.java

```diff
@@ -0,0 +1,42 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Uses UserService to logout the current user and foward them to a desired page
+ */
+@WebServlet("/logout")
+public class LogoutServlet extends HttpServlet {
+  private static final UserService userService = UserServiceFactory.getUserService();
+  /**
+   * Redirects to a page which logs out the user and redirects to the destination-url paramater
+   * If destination-url is not given as a paramater its value defaults to the index
+   * If the user is not logged in then redirect the user directly to the destination-url
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String destinationURL = getDestinationUrl(request);
+    String redirectURL;
+    if(userService.isUserLoggedIn()) {
+      redirectURL = userService.createLogoutURL(destinationURL);
+    } else {
+      redirectURL = destinationURL;
+    }
+    response.sendRedirect(redirectURL);
+  }
+
+  private String getDestinationUrl(HttpServletRequest request){
```

Comment:
If we changed LogingServlet.getDestinationUrl to public static member function, then LogoutServlet would also be able to call this method directly, and we wouldn't have to implement it again here.  What do you think about that approach, would that remove this?

Reply:
I think that's better than what I have right now.

How would an approach like making a utility class for getting parameters from requests be?

Reply:
That would be even better.